MarsRover/Controller.cs

```diff
@@ -113,23 +78,49 @@ private Direction InitialiseDirection()
         private Location InitialiseLocation()
         {
             _output.WriteLine(Messages.RoverStartLocation);
-            var input = _input.Read();
+            var input = _input.ReadLine();
             var isValidLocation = Validator.IsValidLocation(input, Map.Width, Map.Height);
             while(!isValidLocation)
             {
                 _output.WriteLine(Messages.InvalidInput);
                 _output.WriteLine(Messages.RoverStartLocation);
-                input = _input.Read();
+                input = _input.ReadLine();
                 isValidLocation = Validator.IsValidLocation(input, Map.Width, Map.Height);
             }
             var location = InputParser.ParseLocation(input);
-            // need to check if location contains an obstacle
-            if (Map.HasObstacle(location)) // TODO: is this the correct way to do this??
+
+            if (Map.HasObstacle(location)) 
             {
                 _output.WriteLine(Messages.InvalidInput);
                 InitialiseLocation();
             }
             return location;
         }
+        // user asked for string of commands
+        // user enters string of commands
+        // commands are validated
+        // commands are parsed 
+        // rover follows each command
+        // Move logic:
+        // 1/ GetTargetLocation to determine where Rover should Move to
+        // 2a/ use Map.GetSquareAtLocation in order to find Square -  ?? HasObstacle - Controller
+        // 2b/ use Map.HasObstacle on the square to check if there is an obstacle - Controller
+        // 3/ if no obstacle, set Rover.Location to the location of that square - ?? Move
+        // if there is an obstacle, Rover.Location remains the same
+        // Rover reports to user re obstacle
+        public void Run()
+        {
+            _output.WriteLine(Messages.RoverCommands);

```

Comment:
Perhaps the `Setup` method should be called as part of `Run` such that the only public method is `Run`